### PR TITLE
Adds Guest Communicator

### PR DESCRIPTION
**Pull request by richarddubay**
### What does this PR do, or why is it needed?
This PR adds guest communicator abilities to the API and adds the addition of the guest communicator to the sermon notes.

### How do I test this PR?
Make sure all the sermons in the Elephant in the Room series work correctly and there are no errors.
Make sure that Miles McPherson's name shows up in the notes for The Third Option.

## TODO

- [x] I am affirming this is my _best_ work ([Ecclesiastes 9:10](https://www.bible.com/bible/97/ECC.9.10.MSG))
- [x] PR has a relevant title that will be understandable in a public changelog (ie...non developers)
- [x] No new warnings in tests, in storybook, and in-app
- [x] Upload GIF(s) of iOS and Android if applicable
- [x] Set a relevant reviewer

## REVIEW

- [ ] Review updates to test coverage and snapshots
- [ ] Review code through the lens of being concise, simple, and well-documented

**Manual QA**

- [ ] Manual QA on iOS and ensure it looks/behaves as expected
- [ ] Manual QA on Android and ensure it looks/behaves as expected

---

> The purpose of PR Review is to _improve the quality of the software._

### CODE REVIEW

**Review by redreceipt — CHANGES_REQUESTED** [review]
discussed changes

**Comment by richarddubay** [comment]
@redreceipt So I did extract out the `matrixItems` function to it's own function, but I didn't put it in it's own folder yet. Not sure if that's _really_ necessary. But ... I feel good about what's been done here!

**Comment by richarddubay** [comment]
I looked at this a little bit more tonight. I'm not sure how to handle things like `this.request` in another `data-source` file. I might be able to figure it out, but it will probably take me longer than we have. If you know how to do that @redreceipt, I'm open to learning!

File: packages/apollos-church-api/src/data/content-items/data-source.js [hunk 1]
```diff
@@ -8,14 +8,19 @@ import { createAssetUrl } from '../utils';
 const { ROCK, ROCK_CONSTANTS } = ApollosConfig;
 
 export default class ContentItem extends oldContentItem.dataSource {
+  // getMatrixItems = async (matrixItemGuid) => {
+  //   const matrixItems = this.request('/AttributeMatrixItems')
+  //     .filter(`AttributeMatrix/Guid eq guid'${matrixItemGuid}'`)
+  //     .get();
+  //   return matrixItems;
+  // };
+
```
Comment (redreceipt) [inline packages/apollos-church-api/src/data/content-items/data-source.js]: don't need these

File: packages/apollos-church-api/src/data/content-items/data-source.js [hunk 2]
```diff
@@ -8,14 +8,19 @@ import { createAssetUrl } from '../utils';
...
   getContentItemScriptures = async ({ value: matrixItemGuid }) => {
     const {
-      dataSources: { Scripture },
+      dataSources: { Scripture, MatrixItem },
     } = this.context;
     if (!matrixItemGuid) return null;
-    const matrixItems = await this.request('/AttributeMatrixItems')
-      .filter(`AttributeMatrix/Guid eq guid'${matrixItemGuid}'`)
-      .get();
+    const matrixItems = await MatrixItem.getMatrixItems(matrixItemGuid);
```
Comment (redreceipt) [inline packages/apollos-church-api/src/data/content-items/data-source.js]: this function should prob be `getItemsFromGuid`

File: packages/apollos-church-api/src/data/content-items/data-source.js [hunk 3]
```diff
@@ -238,19 +243,37 @@ export default class ContentItem extends oldContentItem.dataSource {
...
+      } = item;
+      if (personAliasGuid !== '') {
+        const { personId } = await this.request('/PersonAlias')
+          .filter(`Guid eq guid'${personAliasGuid}'`)
+          .first();
+        return this.context.dataSources.Person.getFromId(personId);
+      }
+      return null;
+    });
+    return communicators;
```
Comment (redreceipt) [inline packages/apollos-church-api/src/data/content-items/data-source.js]: don't need this extra step, just return `Promise.all(items.map...`

File: packages/apollos-church-api/src/data/content-items/data-source.js [hunk 4]
```diff
@@ -238,19 +243,37 @@ export default class ContentItem extends oldContentItem.dataSource {
...
-    const matrixItem = await this.request('/AttributeMatrixItems')
-      .filter(`AttributeMatrix/Guid eq guid'${matrixItemGuid}'`)
-      .first();
-    if (!matrixItem) return null;
+    const matrixItems = await MatrixItem.getMatrixItems(matrixItemGuid);
+    const communicators = await matrixItems.map(async (item) => {
+      const {
+        attributeValues: { communicator: { value: personAliasGuid } = {} } = {},
+      } = item;
+      if (personAliasGuid !== '') {
```
Comment (redreceipt) [inline packages/apollos-church-api/src/data/content-items/data-source.js]:
```suggestion
      if (personAliasGuid === '') return null
```

File: packages/apollos-church-api/src/data/content-items/data-source.js [hunk 5]
```diff
@@ -238,19 +243,37 @@ export default class ContentItem extends oldContentItem.dataSource {
...
+        const { personId } = await this.request('/PersonAlias')
+          .filter(`Guid eq guid'${personAliasGuid}'`)
+          .first();
+        return this.context.dataSources.Person.getFromId(personId);
+      }
```
Comment (redreceipt) [inline packages/apollos-church-api/src/data/content-items/data-source.js]:
```suggestion
```

File: packages/apollos-church-api/src/data/content-items/data-source.js [hunk 6]
```diff
@@ -238,19 +243,37 @@ export default class ContentItem extends oldContentItem.dataSource {
...
+      return null;
```
Comment (redreceipt) [inline packages/apollos-church-api/src/data/content-items/data-source.js]:
```suggestion
```

File: packages/apollos-church-api/src/data/content-items/data-source.js [hunk 7]
```diff
@@ -238,19 +243,37 @@ export default class ContentItem extends oldContentItem.dataSource {
...
-      .first();
-    return this.context.dataSources.Person.getFromId(personId);
+      dataSources: { MatrixItem },
+    } = this.context;
+    if (!matrixItemGuid) return null;
+    const matrixItems = await MatrixItem.getMatrixItems(matrixItemGuid);
+    const guestCommunicators = await matrixItems.map(
+      (item) => item.attributeValues.guestCommunicator.value
+    );
+    return guestCommunicators;
```
Comment (redreceipt) [inline packages/apollos-church-api/src/data/content-items/data-source.js]: same as above, just return `Promise.all(items.map...`

File: packages/apollos-church-api/src/data/matrix-items/data-source.js [hunk 8]
```diff
@@ -0,0 +1,12 @@
+import RockApolloDataSource from '@apollosproject/rock-apollo-data-source';
+
+export default class MatrixItem extends RockApolloDataSource {
+  expanded = true;
+
+  getMatrixItems = async (matrixItemGuid) => {
+    const matrixItems = await this.request('/AttributeMatrixItems')
```
Comment (redreceipt) [inline packages/apollos-church-api/src/data/matrix-items/data-source.js]: return `this.request...`

**Review by redreceipt — APPROVED** [review]

**Pull request merged**